test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -2016,3 +2016,22 @@ def build(self):
     assert 'DEFINE conan_test_answer=456!' in c.out
     assert 'DEFINE conan_test_other=abc!' in c.out
     assert 'DEFINE conan_test_complex="1 2"!' in c.out
+
+
+@pytest.mark.tool("cmake")
+@pytest.mark.skipif(platform.system() != "Darwin", reason="Only needs to run once, no need for extra platforms")
+def test_cxx_version_not_overriden_if_hardcoded():
+    """Any C++ standard set in the CMakeLists.txt will have priority even if the
+    compiler.cppstd is set in the profile"""
+    tc = TestClient()
+    tc.run("new cmake_exe -dname=foo -dversion=1.0")
+
+    cml_contents = tc.load("CMakeLists.txt")
+    cml_contents = cml_contents.replace("project(foo CXX)\n", "project(foo CXX)\nset(CMAKE_CXX_STANDARD 17)\n")
```

Comment:
I have some questions (maybe for @jcar87), if we made CMAKE_CXX_STANDARD in the ``conan_toolchain.cmake`` to be CACHE variable, will it have precedence over a normal project like this one? 
If we changed it to be a CACHE variable, could it break someone?

Reply:
If `CMAKE_CXX_STANDARD` is defined as a regular (non-cache) variable _after_ the call to `project()` which loads the toolchain - then that one will take precedence, even if we moved to define it as a cache variable ourselves.

Defining it as a cache variable would work if the project files also set a cache variable, in which case the first value would work.

For what's it worth, what's recommended in [Professional CMake ](https://crascit.com/professional-cmake/) is to move away from setting a value and assume it may be set externally:
```
if (NOT DEFINED CMAKE_CXX_STANDARD)
     set(CMAKE_CXX_STANDARD 20)
endif()
```

or if a minimum is required:
```
if (DEFINED CMAKE_CXX_STANDARD)
    if(CMAKE_CXX_STANDARD EQUAL 98 OR CMAKE_CXX_STANDARD LESS 20)
        message(FATAL_ERROR "xxx")
    endif()
else()
    set(CMAKE_CXX_STANDARD 20)
endif()
```

Reply:
I was just wondering if it was worth changing something to deal with projects that didn't follow the recommended practice without needing to patch the CMakeLists.txt, sounds that the ``if()`` recommendation would be the way to go, this PR can be merged.